src/js/App.js

```diff
@@ -1,16 +1,22 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import { HashRouter, Route } from 'react-router-dom';
+import { HashRouter, Route, Redirect, Switch} from 'react-router-dom';
 import Layout from './pages/Layout';
 import ClientMirror from './pages/ClientMirror';
 import ServerMirror from './pages/ServerMirror';
 
 ReactDOM.render(
   <HashRouter>
-    <Layout>
-      <Route exact path="/" component={ClientMirror} />
-      <Route path="/server-mirror" component={ServerMirror} />
-    </Layout>
-  </HashRouter>,
+      <Layout>
+        <Switch>
+          <Redirect exact from="/" to="client-mirror" />
+          <Route path="/client-mirror/:mirrorId?" component={ClientMirror} />
+          <Route path="/server-mirror/:mirrorId?" component={ServerMirror} />
+        </Switch>
```

Comment:
If you don't use `<Switch>` then you can't use `exact` in the redirect, which leads to things things `/client-mirror/` redirecting to `/client-mirror/client-mirror` and so and and so forth for eternity :|
https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/Switch.md